### PR TITLE
harvester: Fix deadlock on disconnect after a repeated plot sync start

### LINKED ISSUE #11466: After upgrading 1.3.4,  harvester suddenly stops working after a period of time
After upgrading version 1.3.4, harvester suddenly stops working after a period of time, never happened in versions before 1.3.4.
the chia process status is normal and not exited
The network is fine, but the connection between harvester and farmer has been disconnected and will not reconnect automatically.
It did not return to normal until the harvester was restarted.

farmer/harvester os: ubuntu linux 20.04
```
2022-05-06T23:52:20.418 harvester chia.harvester.harvester: INFO     1 plots were eligible for farming 5891ffa946... Found 0 proofs. Time: 0.19518 s. Total 1043 plots
2022-05-06T23:52:21.411 harvester chia.harvester.harvester: INFO     _plot_refresh_callback: event done, loaded 0, removed 0, processed 1043, remaining 0, duration: 0.15 seconds, total plots: 1043
2022-05-06T23:52:31.748 harvester chia.plot_sync.sender   : INFO     _send_next_message didn't receive response expected_message_type: plot_sync_removed, expected_identifier: {'message_id': 2, 'sync_id': 1651852341, 'timestamp': 1651852341}, message None
2022-05-06T23:52:51.811 harvester chia.plot_sync.sender   : INFO     _send_next_message didn't receive response expected_message_type: plot_sync_removed, expected_identifier: {'message_id': 2, 'sync_id': 1651852341, 'timestamp': 1651852361}, message None
2022-05-06T23:53:11.871 harvester chia.plot_sync.sender   : INFO     _send_next_message didn't receive response expected_message_type: plot_sync_removed, expected_identifier: {'message_id': 2, 'sync_id': 1651852341, 'timestamp': 1651852381}, message None
2022-05-06T23:53:31.930 harvester chia.plot_sync.sender   : INFO     _send_next_message didn't receive response expected_message_type: plot_sync_removed, expected_identifier: {'message_id': 2, 'sync_id': 1651852341, 'timestamp': 1651852401}, message None
2022-05-06T23:53:51.992 harvester chia.plot_sync.sender   : INFO     _send_next_message didn't receive response expected_message_type: plot_sync_removed, expected_identifier: {'message_id': 2, 'sync_id': 1651852341, 'timestamp': 1651852421}, message None
2022-05-06T23:54:12.052 harvester chia.plot_sync.sender   : INFO     _send_next_message didn't receive response expected_message_type: plot_sync_removed, expected_identifier: {'message_id': 2, 'sync_id': 1651852341, 'timestamp': 1651852441}, message None
2022-05-06T23:54:21.679 harvester harvester_server        : INFO     Connection closed: 192.168.137.137, node id: 6e88f6d4a23d14db8003ceccc1acb1248c150ec4eb27308b7052b3b8d5e19385
2022-05-06T23:54:21.705 harvester chia.harvester.harvester: INFO     peer disconnected {'host': '192.168.137.137', 'port': 8447}

------------There are no logs after this-------------------
```

The same phenomenon as the following post

### Discussed in https://github.com/Chia-Network/chia-blockchain/discussions/11326

<div type='discussions-op-text'>

<sup>Originally posted by **tonyx888** April 27, 2022</sup>
Hello everyone, 
since I updated to version 1.3.4 I have many problems with harvesters, 
I have a network of 26 harvesters and 1 farmer all updated to version 1.3.4 
Unfortunately my internet connection every day falls 1 or 2 times, before with 1.2.11 the harvesters reconnected immediately, now with the latest version of the harvesters on the log I have this error and loses the connection forcing me to have to manually restart the harvesters.
```
2022-04-26T16:34:57.006 harvester harvester_server        : INFO     Cannot connect to host 192.168.0.5:8447 ssl:<ssl.SSLContext object at 0x000001A4029977C0> [Periodo di timeout del semaforo scaduto]
2022-04-26T16:35:00.021 harvester harvester               : INFO     Reconnecting to peer {'host': '192.168.0.5', 'port': 8447}
2022-04-26T16:35:07.113 harvester harvester_server        : INFO     Connected with farmer {'host': '192.168.0.5', 'port': 8447}
2022-04-26T16:35:07.238 harvester chia.plotting.manager   : INFO     Loaded 534096 bytes of cached data
2022-04-26T16:35:35.372 harvester chia.plot_sync.sender   : INFO     _send_next_message didn't receive response expected_message_type: plot_sync_loaded, expected_identifier: {'message_id': 7, 'sync_id': 1650983707, 'timestamp': 1650983725}, message None
2022-04-26T16:35:55.446 harvester chia.plot_sync.sender   : INFO     _send_next_message didn't receive response expected_message_type: plot_sync_loaded, expected_identifier: {'message_id': 7, 'sync_id': 1650983707, 'timestamp': 1650983745}, message None
2022-04-26T16:36:15.504 harvester chia.plot_sync.sender   : INFO     _send_next_message didn't receive response expected_message_type: plot_sync_loaded, expected_identifier: {'message_id': 7, 'sync_id': 1650983707, 'timestamp': 1650983765}, message None
2022-04-26T16:36:35.577 harvester chia.plot_sync.sender   : INFO     _send_next_message didn't receive response expected_message_type: plot_sync_loaded, expected_identifier: {'message_id': 7, 'sync_id': 1650983707, 'timestamp': 1650983785}, message None
2022-04-26T16:36:55.651 harvester chia.plot_sync.sender   : INFO     _send_next_message didn't receive response expected_message_type: plot_sync_loaded, expected_identifier: {'message_id': 7, 'sync_id': 1650983707, 'timestamp': 1650983805}, message None
2022-04-26T16:37:15.708 harvester chia.plot_sync.sender   : INFO     _send_next_message didn't receive response expected_message_type: plot_sync_loaded, expected_identifier: {'message_id': 7, 'sync_id': 1650983707, 'timestamp': 1650983825}, message None
2022-04-26T16:37:25.269 harvester harvester_server        : INFO     Connection closed: 192.168.0.5, node id: 098ba0fa3b705a6f8084d5c5869ac64c7c569fd2820a13a9421b15d2d1978c23
2022-04-26T16:37:25.269 harvester chia.harvester.harvester: INFO     peer disconnected {'host': '192.168.0.5', 'port': 8447}
```

PLEASE HELP ME
Thanks all</div>

### PR DESCRIPTION
**Pull request by xdustinface**
Currently on main if the farmer/harvester connections drops after the farmer didn't respond to plot sync messages for more than the refresh interval of the harvester we end up with a harvester deadlock because the plot manager thread stucks here https://github.com/Chia-Network/chia-blockchain/blob/a4ec7e7cb8d8ecfa2a50c038e5dd94ef3e620c6c/chia/plot_sync/sender.py#L259

while the main threads wants to join it here as part of the disconnect https://github.com/Chia-Network/chia-blockchain/blob/a4ec7e7cb8d8ecfa2a50c038e5dd94ef3e620c6c/chia/harvester/harvester.py#L107

See the test b49ef1640b52ce29c019e0e7488dcceca9ce94b5 which fails without the other two commits.

### CODE REVIEW

**Review by altendky — DISMISSED** [review]

File: chia/plot_sync/sender.py [hunk 1]
```diff
@@ -297,10 +297,9 @@ def _finalize_sync(self) -> None:
         self._sync_id = uint64(0)
```
[the remark this answers is not in the document]
Comment (xdustinface) [reply chia/plot_sync/sender.py]: Good point, see [4c2dcb9](https://github.com/Chia-Network/chia-blockchain/pull/11481/commits/4c2dcb98ff8692cf3a437df14b4878a0dbdc0497)

**Review by altendky — APPROVED** [review]

**Comment by pnoch** [comment]
should this be included in to 1.3.5 release as well?

**Comment by pnoch** [comment]
Tested on my 5p plots miner, very nice.

Great work! @xdustinface

**Comment by yang1782** [comment]
All of my harvesters stopped connecting today on Windows. Restarting and upgrading to 1.3.5 did not help. I will try to copy the certificates from the farmer again to see if that helps.

**Comment by rluisr** [comment]
same here on 1.3.5

**Comment by xdustinface** [comment]
This PR isn't included in 1.3.5.

**Pull request merged**